unittests/test_utility.py

```diff
@@ -621,62 +625,61 @@ def test_key_resolution(self):
-        self.assertEqual(4, scoped_dict['a:b:k3'])
-        self.assertEqual(10, scoped_dict['a:b:k4'])
+        assert 3 == scoped_dict['a:b:k1']
+        assert 2 == scoped_dict['a:b:k2']
+        assert 4 == scoped_dict['a:b:k3']
+        assert 10 == scoped_dict['a:b:k4']
 
-        self.assertEqual(3, scoped_dict['a:b:c:k1'])
-        self.assertEqual(5, scoped_dict['a:b:c:k2'])
-        self.assertEqual(6, scoped_dict['a:b:c:k3'])
-        self.assertEqual(10, scoped_dict['a:b:c:k4'])
+        assert 3 == scoped_dict['a:b:c:k1']
+        assert 5 == scoped_dict['a:b:c:k2']
+        assert 6 == scoped_dict['a:b:c:k3']
+        assert 10 == scoped_dict['a:b:c:k4']
 
         # Test global scope
-        self.assertEqual(7, scoped_dict['k1'])
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['k2']", globals(), locals()
-        )
-        self.assertEqual(9, scoped_dict['k3'])
-        self.assertEqual(10, scoped_dict['k4'])
-
-        self.assertEqual(7, scoped_dict[':k1'])
-        self.assertRaises(
-            KeyError, exec, "scoped_dict[':k2']", globals(), locals()
-        )
-        self.assertEqual(9, scoped_dict[':k3'])
-        self.assertEqual(10, scoped_dict[':k4'])
-
-        self.assertEqual(7, scoped_dict['*:k1'])
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['*:k2']", globals(), locals()
-        )
-        self.assertEqual(9, scoped_dict['*:k3'])
-        self.assertEqual(10, scoped_dict['*:k4'])
+        assert 7 == scoped_dict['k1']
+        with pytest.raises(KeyError):
+            exec("scoped_dict['k2']", globals(), locals())
+
+        assert 9 == scoped_dict['k3']
+        assert 10 == scoped_dict['k4']
+
+        assert 7 == scoped_dict[':k1']
+        with pytest.raises(KeyError):
+            exec("scoped_dict[':k2']", globals(), locals())
+
+        assert 9 == scoped_dict[':k3']
+        assert 10 == scoped_dict[':k4']
+
+        assert 7 == scoped_dict['*:k1']
+        with pytest.raises(KeyError):
+            exec("scoped_dict['*:k2']", globals(), locals())
+
+        assert 9 == scoped_dict['*:k3']
+        assert 10 == scoped_dict['*:k4']
 
         # Try to fool it, by requesting keys with scope names
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['a']", globals(), locals()
-        )
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['a:b']", globals(), locals()
-        )
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['a:b:c']", globals(), locals()
-        )
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['a:b:c:d']", globals(), locals()
-        )
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['*']", globals(), locals()
-        )
-        self.assertRaises(
-            KeyError, exec, "scoped_dict['']", globals(), locals()
-        )
+        with pytest.raises(KeyError):
+            exec("scoped_dict['a']", globals(), locals())
```

Comment:
Replace all `exec` statements here:

```suggestion
            scoped['a']
```